lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -45,12 +58,165 @@ namespace {
   // once per top-level file read operation (ie, once per open, or once
   // per import, or once per reference).
   std::atomic<size_t> readDepth;
+
+#if defined(WANT_UFE_BUILD)
+  // The proxy shape has an internal cache which needs to update when any of
+  // its UFE scene items are selected and transformed.
+  class UfeTransformObserver : public Ufe::Observer
+  {
+  public:
+    UfeTransformObserver() : Ufe::Observer()
+    {
+    }
+
+    ~UfeTransformObserver() override
+    {
+    }
+
+    void operator()(const Ufe::Notification& notification) override
+    {
+      auto xformChanged = dynamic_cast<const Ufe::Transform3dChanged*>(&notification);
+      if (xformChanged == nullptr) return;
+
+      // Action for USD scene items only.
+      Ufe::SceneItem::Ptr sceneItem = xformChanged->item();
+      if (!sceneItem || (sceneItem->runTimeId() != AL::usdmaya::USD_UFE_RUNTIME_ID)) return;
+
+      std::string mayaPath = sceneItem->path().popSegment().popHead().string();
+
+      MSelectionList sl;
+      sl.add(MString(mayaPath.c_str()));
```

Comment:
```c++
MString(mayaPath.c_str(), mayaPath.length())
```
since you know the length in advance, may as well use it (avoids an extra strclen)

Reply:
Agreed. Change made.